Allow manual ordering on joined tables and/or aliased columns

retorted opened the pull request:
~~I ran into some cases where `order_manual()` isn't quite enough flexibility, as it automatically assigns the table alias.~~
~~What do you think to `order_by_raw()`?~~

Updated purpose/scope; changes to `order_manual()`:
- `order_manual()` now uses `&'static` for it's sql argument
- `order_manual()` no longer inserts the table alias by default
- `order_manual()` now interpolates `$` as the table alias, like similar functions (`set_manual()`, etc)
- `order_manual()` is also added to the `SelectBuilder` interface (previously was only available in `QueryBuilder`)
- test added for `order_manual()` output, with three different syntax examples

lex148 commented:
I think you just found two bugs. 

I would not have expected order_manual to automatically prepend the table alias. 
I would have expected it to use the same `$` syntax as the other *_manual functions to refer to the current table. 

Additionally This really needs to be `&'static` to help prevent SQL injection. Steering users to SQL parameters is a really good thing.

retorted commented:
Sounds good to me :+1:

retorted commented:
Updated, summary: 
- `order_manual()` now uses `&'static` for it's sql argument
- `order_manual()` now interpolates `$` as the table alias, like similar functions (set_manual, etc)
- `order_manual()` is also added to the `SelectBuilder` interface (previously was only available in `QueryBuilder`)
- tests added for `order_manual()` output, with three different cases

retorted commented:
As a happy side-effect to these changes, if someone is building a group by query with aggregate functions, they can now order the results by the aggregated values, eg:
```rust
Team::all()
    .select(|t| t.name)
    .left_join(|t| t.players, {
        Player::all().select_count(|p| p.id, "player_count")
    })
    .group_by(|t| t.name)
    .order_manual("player_count DESC") // <---
```

lex148 commented:
This is great thanks !
